TESTING.md

```diff
@@ -54,16 +54,50 @@ property or `MANTA_IT_NO_CLEANUP` environment variable.
 
 # Running Tests
 
-Run `mvn verify` from the project root to run all tests. Some Maven goals will
-not work
+Run `mvn verify` from the project root to run all tests. Some Maven goals will not work.
 
 While the Java Cryptography Extensions are expected to be installed, it is
-possible torun a subset of the test suite by adding
-`-DexcludedGroups=unlimited-crypto`, e.g.:
+possible to run a subset of the test suite by adding
+`-DexcludedGroups=unlimited-crypto` or `-Dgroups=headers,error`, e.g.
 ```
 mvn test -DexcludedGroups=unlimited-crypto
+mvn verify -Dgroups=headers,error
 ```
 
+Also, if we are running a defined TestNG group, invoke
+`-Dgroups=headers`, e.g:
+
+```
+mvn clean verify -Dgroups=headers
+```
+
+Note: Since we have transitioned from wildcard to explicit test definition, it is **essential**
+for engineers to add integration-tests in `testng-it.xml` file complying with the nature of their 
+corresponding TestNG groups. For instance, if we create a new integration-test `ExampleHttpHeadersIT.java` 
+for `headers`, that test will be defined under:
+```
+    <test name="Manta Client Http Headers Tests">
+        <groups>
+            <define name="headers" />
+        </groups>
+        <classes>
+            <class name="com.joyent.manta.http.MantaHttpHeadersIT" />
+            <class name="com.joyent.manta.http.ExampleHttpHeadersIT" />
+        </classes>
+    </test>
+```
+
+## Dry run
+	
+The following invocation will print a list of which integration tests would run and their relevant
+parameters::
+	
+```
+mvn verify -Dit.dryRun=true
+```
+	
+Note: This has not yet been expanded to unit tests.
```

Comment:
You're absolutely correct this won't be incorporated into unit-tests. I merely stated it so that the SDK user doesn't erroneously assume that `dry-Run` is available as a feature for unit-tests too.
Maybe this sentence should be re-edited from
`Note: This has not yet been expanded to unit tests.` -> `Note: This feature is invalid for unit tests.`